src/operations/transforms/rotate.js

```diff
@@ -0,0 +1,84 @@
+const flatten = require('../../utils/flatten')
+
+const mat4 = require('../../math/mat4')
+
+const {geom2, geom3, path2} = require('../../geometry')
+
+// Tait-Bryan Euler angle convention using active, intrinsic rotations around the axes in the order z-y-x
+// angle of rotations
+//   yaw/Z rotation
+//   pitch/Y rotation
+//   roll/X rotation
+const rotation = (yaw, pitch, roll) => {
+  // convert to radians
+  yaw = yaw * Math.PI * (1.0 / 180.0)
+  pitch = pitch * Math.PI * (1.0 / 180.0)
+  roll = roll * Math.PI * (1.0 / 180.0)
+
+  // precompute sines and cosines of Euler angles
+  const sy = Math.sin(yaw);
+  const cy = Math.cos(yaw);
+  const sp = Math.sin(pitch);
+  const cp = Math.cos(pitch);
+  const sr = Math.sin(roll);
+  const cr = Math.cos(roll);
+
+  // create and populate rotation matrix
+  // clockwise rotation
+  //const els = [
+  //  cp*cy, sr*sp*cy - cr*sy, sr*sy + cr*sp*cy, 0,
+  //  cp*sy, cr*cy + sr*sp*sy, cr*sp*sy - sr*cy, 0,
+  //  -sp, sr*cp, cr*cp, 0,
+  //  0, 0, 0, 1
+  //]
+  // counter clockwise rotation
```

Comment:
Matrix `rotZ = rotation(90, 0, 0)` corresponding to code below this comment corresponds to clockwise rotation of an object. It transforms vector [1, 0, 0] to [0, -1, 0] rather than [0, 1, 0]. There are two misconceptions are possible

1) What is rotated an object or a reference frame? The matrix is applied to objects.
2) What type of coordinate system is used left or right?

I think, this information should be reflected in documentation.

I did use this simple test to validate correctness of rotation
```javascript
/** Performs 4x4 matrix dot product with 4x1 vector */
function dot(mat, vect) {
  // assuming the matrixs is defined by rows
  const dotRow = (m) => [0,1,2,3].map( n => [n, 4 * m + n] )
                                 .reduce( (acc, n) => acc + mat[n[1]] * vect[n[0]], 0.0);
  return [0,1,2,3].map(dotRow);
}

function assertEqualVectors(title, actual, expected) {
  
  const tolerance = 1e-14;  // for comparison double precision values
  const isEqualVectors = [0,1,2,3].reduce((acc, n) => acc && Math.abs(actual[n] - expected[n]) < tolerance, true);

  if (isEqualVectors) {
    console.log('PASS - ' + title);
  } else {
    console.log('FAIL - ' + title);
    console.log(`    actual: [${actual}]`);
    console.log(`  expected: [${expected}]`);
  }
  console.log();
}

const unitX = [1, 0, 0, 1];
const unitY = [0, 1, 0, 1];
const unitZ = [0, 0, 1, 1];

const rotZ = rotation(90, 0,  0);
const rotX = rotation( 0, 0, 90);

assertEqualVectors('rotZ X->Y', dot(rotZ, unitX), unitY); // fail because of [0,-1,0,1]
assertEqualVectors('rotZ Z->Z', dot(rotZ, unitZ), unitZ);

assertEqualVectors('rotX Y->Z', dot(rotX, unitY), unitZ); // fail because of [0,0,-1,1]
assertEqualVectors('rotX X->X', dot(rotX, unitX), unitX);
```

Reply:
@vr050714 thanks for the comments. i feel slightly more confident now.

I think the rotation() function should be moved out, and added as mat4.fromTaitBryanAngles()
I can add tests based on the code from @vr050714

@kaosat-dev does this sound fine?

Reply:
@z3dev that sound like a *very* good idea ! 
Thanks for the very good feedback  @vr050714 !